osquery/tables/system/linux/disk_encryption.cpp

```diff
@@ -0,0 +1,100 @@
+#include <unistd.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/sql.h>
+
+extern "C" {
+#include <libcryptsetup.h>
+
+crypt_status_info crypt_status(struct crypt_device *cd, const char *name);
+
+#ifdef CENTOS_CENTOS6
+int crypt_init_by_name(struct crypt_device **cd, const char *name);
+#define crypt_init(cd, name) crypt_init_by_name(&cd, name.c_str())
```

Comment:
What do you think of:

```cpp
#define crypt_init(cd, name) crypt_init_by_name(cd, name)
// call: crypt_init(&cd, name.c_str());
```

Then we can keep the c-isms in the code vs. define/macros.